From 15f6bb17257a42176aac013342f9359ee432dd1f Mon Sep 17 00:00:00 2001
From: Hu Kejun <william.hu@rock-chips.com>
Date: Sun, 13 Dec 2020 11:02:35 +0800
Subject: [PATCH] media: i2c: restore vcm_config ioctl

restore vcm_config ioctl deleted by commit a3907e42d201 ("media: add motor driver of MP6507 for camera IRIS/FOCUS/ZOOM")

Signed-off-by: Hu Kejun <william.hu@rock-chips.com>
Change-Id: I391ec77a31291c65fbb04da1915647be25dcf4b5
---
 drivers/media/i2c/rk_vcm_head.h | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/drivers/media/i2c/rk_vcm_head.h b/drivers/media/i2c/rk_vcm_head.h
index cf49fa237e86..5465337545e1 100644
--- a/drivers/media/i2c/rk_vcm_head.h
+++ b/drivers/media/i2c/rk_vcm_head.h
@@ -26,12 +26,17 @@
 #define RK_VIDIOC_ZOOM_TIMEINFO \
 	_IOR('V', BASE_VIDIOC_PRIVATE + 2, struct rk_cam_vcm_tim)
 
+#define RK_VIDIOC_GET_VCM_CFG \
+	_IOR('V', BASE_VIDIOC_PRIVATE + 3, struct rk_cam_vcm_cfg)
+#define RK_VIDIOC_SET_VCM_CFG \
+	_IOW('V', BASE_VIDIOC_PRIVATE + 4, struct rk_cam_vcm_cfg)
+
 #define RK_VIDIOC_FOCUS_CORRECTION \
-	_IOR('V', BASE_VIDIOC_PRIVATE + 3, unsigned int)
+	_IOR('V', BASE_VIDIOC_PRIVATE + 5, unsigned int)
 #define RK_VIDIOC_IRIS_CORRECTION \
-	_IOR('V', BASE_VIDIOC_PRIVATE + 4, unsigned int)
+	_IOR('V', BASE_VIDIOC_PRIVATE + 6, unsigned int)
 #define RK_VIDIOC_ZOOM_CORRECTION \
-	_IOR('V', BASE_VIDIOC_PRIVATE + 5, unsigned int)
+	_IOR('V', BASE_VIDIOC_PRIVATE + 7, unsigned int)
 
 #ifdef CONFIG_COMPAT
 #define RK_VIDIOC_COMPAT_VCM_TIMEINFO \
-- 
2.35.3

